AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java

```diff
@@ -0,0 +1,175 @@
+import com.ooyala.android.item.Stream;
+import com.ooyala.android.player.StreamPlayer;
+import com.ooyala.android.util.DebugMode;
+
+import java.util.Set;
+import java.util.Timer;
+import java.util.TimerTask;
+
+
+/**
+ *  This is a new player which extends from StreamPlayer.
+ *  Extending from StreamPlayer is necessary to get it work with Ooyala SDk.
+ *
+ */
+public class SampleVideoStreamPlayer extends StreamPlayer {
+
+    private static final String TAG = SampleVideoStreamPlayer.class.getSimpleName();
+
+    private int DURATION;
+    private final int REFRESH_RATE = 250;
+
+    private Timer timer;
+    private int playhead = 0;
+    private Handler timerHandler;
+    private TextView textView;
+    private LinearLayout linearLayout;
+
+    private Stream stream;
+
+
+    @Override
+    public void init(OoyalaPlayer parent, Set<Stream> streams) {
+        WifiManager wifiManager = (WifiManager)parent.getLayout().getContext().getSystemService(Context.WIFI_SERVICE);
+        boolean isWifiEnabled = wifiManager.isWifiEnabled();
+
+        stream = Stream.bestStream(streams, isWifiEnabled);
+        if (stream == null) {
+            DebugMode.logE(TAG, "ERROR: Invalid Stream (no valid stream available)");
+            this._error = new OoyalaException(OoyalaException.OoyalaErrorCode.ERROR_PLAYBACK_FAILED, "Invalid Stream");
+            setState(OoyalaPlayer.State.ERROR);
+            return;
+        }
+
+        if (parent == null) {
+            this._error = new OoyalaException(OoyalaException.OoyalaErrorCode.ERROR_PLAYBACK_FAILED, "Invalid Parent");
+            setState(OoyalaPlayer.State.ERROR);
+            return;
+        }
+        DURATION = parent.getCurrentItem().getDuration() != 0 ? parent.getCurrentItem().getDuration() : 15000 ;
+        setParent(parent);
+        initPlayer();
+    }
+
+    public void initPlayer () {
+        // create layout/view which will contain playerview
+        linearLayout = new LinearLayout(_parent.getLayout().getContext());
+        linearLayout.setLayoutParams(new FrameLayout.LayoutParams(
+                ViewGroup.LayoutParams.MATCH_PARENT,
+                ViewGroup.LayoutParams.MATCH_PARENT, Gravity.CENTER));
+        linearLayout.setBackgroundColor(Color.BLACK);
+        textView = new TextView(_parent.getLayout().getContext());
+        linearLayout.addView(textView);
+        // Add newly created playerView/ to parent which is Ooyala player
+        _parent.getLayout().addView(linearLayout);
+        timerHandler = new Handler() {
+            public void handleMessage(Message msg) {
+                refresh();
+            }
+        };
+
+        // if auto-play is enabled the state neeeds to be PLAYING
+        if (_parent.getDesiredState() == OoyalaPlayer.DesiredState.DESIRED_PLAY) {
+          setState(OoyalaPlayer.State.PLAYING);
+        } else {
+          setState(OoyalaPlayer.State.READY);
+        }
+
+    }
+
+    @Override
+    public void resume() {
```

Comment:
To make this cleaner, can you remove the @override methods that aren't doing anything?